W-20668129 | Correct AS2 Connector Documentation (#3106)
mulesoft/docs-connectors: as2/6.9/modules/ROOT/pages/as2-connector-example-receive-send.adoc

[[synchronous-mdns]]
== Sending Synchronous AS2 MDNs

> When your trading partner is using an *AS 2 Listener* (for example, the <<receiving-messages,Receiving AS2 Messages>> example in this page), the listener’s MDN mode controls when a synchronous MDN is returned:

Reviewer: ```suggestion
When your trading partner uses an AS2 Listener, like in the <<receiving-messages-example,Receiving AS2 Messages>> example, the listener’s MDN mode controls when a synchronous MDN is returned. The AS2 Listener supports these configurations for managing synchronous MDN delivery timing:
```

Reviewer: To link directly to the example, I recommend adding [[receiving-messages-example]] above the "XML for the Receiving AS2 Messages Example" heading.

> - IMMEDIATE: The *AS 2 Listener*  sends the synchronous MDN back on the same HTTP connection as soon as the incoming AS2 message is parsed and validated. The receiver’s business flow continues after the MDN is already sent.

Reviewer: ```suggestion
* *Immediate*
+
The listener returns the MDN as soon as the incoming message is parsed and validated. The receiver’s business flow continues as a separate process after the MDN has already been sent.
```

> - AUTO: The *AS 2 Listener*  sends the synchronous MDN back on the same HTTP connection only after its business flow completes. If the flow fails, a negative MDN is returned.

Reviewer: ```suggestion
* *Auto*
+
The listener returns the MDN only after its internal business flow completes. This ensures that if the flow fails, a negative MDN is returned to the sender to signal a processing error.
```

[[asynchronous-mdns]]
== Sending Asynchronous AS2 MDNs

> When receiving an AS2 message that requests an asynchronous MDN, the *AS 2 Listener* validates for presence of `Disposition-Notification-To` and the `Receipt-Delivery-Option` headers. 

Reviewer: ```suggestion

```

> The *AS 2 Listener* immediately returns HTTP 200 OK to acknowledge the inbound AS2 message, and the URL configured in the `Receipt-Delivery-Option` header is used to retun the MDN via a new HTTP connection.

Reviewer: ```suggestion
```

> The listener’s MDN mode controls when a asynchronous MDN is returned::

Reviewer: ```suggestion
The AS 2 Listener validates the presence of the `Disposition-Notification-To` and `Receipt-Delivery-Option` headers when receiving an asynchronous MDN request. The listener immediately returns an `HTTP 200 OK` to acknowledge the inbound message. The listener then uses the URL in the `Receipt-Delivery-Optio`n header to return the MDN via a new HTTP connection.

The AS2 Listener supports these configurations for managing asynchronous MDN delivery timing:
```

> - IMMEDIATE: The *AS 2 Listener*  posts the asynchronous MDN to the `Receipt-Delivery-Option` URL that is sent in header as soon as the inbound message is parsed and validated.

Reviewer: ```suggestion
* *Immediate*
+
The listener posts the MDN to the `Receipt-Delivery-Option` URL as soon as the inbound message is parsed and validated.
```

> - AUTO: The *AS 2 Listener*  posts the asynchronous MDN to the `Receipt-Delivery-Option` URL that is sent in header only after its business flow completes. If the flow fails, a negative MDN is posted.

Reviewer: ```suggestion
* *Auto*
+
The listener posts the MDN to the `Receipt-Delivery-Option` URL only after its business flow completes, allowing it to send a negative MDN if the flow fails.
```

[[synchronous-mdns]]
== Sending Synchronous AS2 MDNs

When your trading partner uses an AS2 listener, like in the <<receiving-messages-example,Receiving AS2 Messages>> example, the listener's MDN mode controls when a synchronous MDN is returned. The AS2 listener supports these configurations for managing synchronous MDN delivery timing:

* *Immediate*
+
The listener returns the MDN as soon as the incoming message is parsed and validated. The receiver's business flow continues as a separate process after the MDN is sent. This is the default MDN Mode. 
* *Auto*
+
The listener returns the MDN only after its internal business flow completes. This ensures that if the flow fails, a negative MDN is returned to the sender to signal a processing error.
+

> For this configuration, the MDN mode must be set to `AUTO` on the AS2 listener.

Reviewer 2: ```suggestion
For this configuration, set the MDN mode to `AUTO` on the AS2 listener.
```
